developer/services/qrexec.md

```diff
@@ -0,0 +1,298 @@
+Therefore the design decision was made to pass all control communication via dom0, that can enforce proper authorization.
+Then, it is natural to reuse the already-existing qrexec framework.
+
+Also, note that bare qrexec provides `VM <-> dom0` connectivity, but the command execution is always initiated by dom0.
+There are cases when VM needs to invoke and send data to a command in dom0 (e.g. to pass information on newly installed `.desktop` files).
+Thus, the framework allows dom0 to be the RPC target as well.
+
+Thanks to the framework, RPC programs are very simple -- both RPC client and server just use their stdin/stdout to pass data.
+The framework does all the inner work to connect these processes to each other via `qrexec-daemon` and `qrexec-agent`.
+Additionally, disposable VMs are tightly integrated -- RPC to a DisposableVM is identical to RPC to a normal domain, all one needs is to pass `@dispvm` as the remote domain name.
+
+## Qubes RPC administration
+
+<!-- (*TODO: fix for non-linux dom0*) -->
+
+### Policy files
+
+The dom0 directory `/etc/qubes-rpc/policy/` contains a file for each available RPC action that a VM might call.
+Together the contents of these files make up the RPC access policy database.
+Policies are defined in lines with the following format:
+
+    srcvm destvm (allow|deny|ask[,default_target=default_target_VM])[,user=user_to_run_as][,target=VM_to_redirect_to]
+
+You can specify srcvm and destvm by name or by one of the reserved keywords such as `@anyvm`, `@dispvm`, or `dom0`.
+(Of these three, only `@anyvm` keyword makes sense in the srcvm field.
+Service calls from dom0 are currently always allowed, and `@dispvm` means "new VM created for this particular request," so it is never a source of request.)
+Other methods using *tags* and *types* are also available (and discussed below).
+
+Whenever a RPC request for an action is received, the domain checks the first matching line of the relevant file in `/etc/qubes-rpc/policy/` to determine access:
+whether to allow the request, what VM to redirect the execution to, and what user account the program should run under.
+Note that if the request is redirected (`target=` parameter), policy action remains the same -- even if there is another rule which would otherwise deny such request.
+If no policy rule is matched, the action is denied.
+If the policy file does not exist, the user is prompted to create one.
+If there is still no policy file after prompting, the action is denied.
+
+In the target VM, the file `/etc/qubes-rpc/RPC_ACTION_NAME` must exist, containing the file name of the program that will be invoked, or being that program itself -- in which case it must have executable permission set (`chmod +x`).
+
+### Making an RPC call
+
+From outside of dom0, RPC calls take the following form:
+
+    $ qrexec-client-vm target_vm_name RPC_ACTION_NAME rpc_client_path client arguments
+
+For example:
+
+    $ qrexec-client-vm work qubes.StartApp firefox
```

Comment:
```suggestion
    $ qrexec-client-vm work qubes.StartApp+firefox
```
Argument to the service is part of service name, with `+` separator. Otherwise it would be parsed as rpm_client_path.

Reply:
:heavy_check_mark: